### PR TITLE
Temporal: Move more ZonedDateTime tests out of staging

### PR DESCRIPTION
**Pull request by catamorphism**

### CODE REVIEW

File: test/built-ins/Temporal/ZonedDateTime/prototype/add/overflow.js
Comment (ptomato) [inline test/built-ins/Temporal/ZonedDateTime/prototype/add/overflow.js:1]: This one should be split into `add/overflow.js` and `subtract/overflow.js`. The `add` and `subtract` parts of the test are pretty much independent from each other.

File: test/built-ins/Temporal/ZonedDateTime/prototype/equals/casts-argument.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2021 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.zoneddatetime.prototype.equals
+description: Different time zones not equal.
```
Comment (ptomato) [inline test/built-ins/Temporal/ZonedDateTime/prototype/equals/casts-argument.js:6]: Wrong description

**Review by ptomato — CHANGES_REQUESTED** [review]
Thanks, one minor comment and one further split

File: test/built-ins/Temporal/ZonedDateTime/prototype/add/overflow.js
Comment (catamorphism) [reply test/built-ins/Temporal/ZonedDateTime/prototype/add/overflow.js:1]: Done

File: test/built-ins/Temporal/ZonedDateTime/prototype/equals/casts-argument.js (hunk 1 above)
Comment (catamorphism) [reply test/built-ins/Temporal/ZonedDateTime/prototype/equals/casts-argument.js]: Fixed

**Review by ptomato — APPROVED** [review]
Thanks!

**Pull request merged**